classes/check/collectorcheck.php

```diff
@@ -0,0 +1,78 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Collector failure tests for Check API
+ *
+ * @package admin-tool_cloudmetrics
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @copyright Catalyst IT
+ */
+
+namespace tool_cloudmetrics\check;
+
+use core\check\check;
+use core\check\result;
+
+class collectorcheck extends check {
+
+    /**
+     * Constructor
+     */
+    public function __construct() {
+        $this->id = 'checkcollectorcheck';
+        $this->name = get_string('checkcollectorcheck', 'tool_cloudmetrics');
+    }
+
+    /**
+     * A link to a place to action this
+     *
+     * @return action_link|null
+     */
+    public function get_action_link(): ?\action_link {
+        return new \action_link(
+            new \moodle_url('/admin/tool/cloudmetrics'),
+            get_string('managelink', 'tool_cloudmetrics'));
+    }
+
+    /**
+     * Return result
+     * @return result
+     */
+    public function get_result() : result {
+
+        $failures = false;
+        $tests = get_config('tool_cloudmetrics_failure');
+        if ($tests) {
+            foreach ($tests as $name => $result) {
+                if ($result === 'pass') {
+                    continue;
+                }
+                $failures = true;
+                $summarytemplate = get_string('collector_failed', 'tool_cloudmetrics', $name);
+                $summary = sprintf($summarytemplate, date('Y-m-d H:i:s', (int) $result));
+                return new result(result::ERROR, $summary, '');
+            }
+
+        }
+
+        if (!$failures) {
+            $summary = get_string('no_issues', 'tool_cloudmetrics');
+            return new result(result::OK, $summary, '');
```

Comment:
Each check has a summary and a details string. In cases like this it is often good to have the summary be good / bad, but the details has a list of each thing it checked and whether it is good or bad.